planner/core/planbuilder.go

```diff
@@ -730,6 +730,10 @@ func (b *PlanBuilder) filterPathByIsolationRead(paths []*util.AccessPath, dbName
 	if dbName.L == mysql.SystemDB {
 		return paths, nil
 	}
+	cfgIsolationEngines := make(map[string]struct{})
+	for _, engine := range config.GetGlobalConfig().IsolationRead.Engines {
+		cfgIsolationEngines[engine] = struct{}{}
```

Comment:
This map can be initialized when we set the config so we don't need to always create a new map when we complie one SQL?

Reply:
For the server config. It's hard to create a map by `toml`.

Reply:
Do you have any advice?

Reply:
Make it a global variable and init it in places around `Config::Valid()` for example?

Reply:
I prefer not to do this. There are two reasons:
1. If we set the map in `Config::Valid()`, there are many places of tests need to keep consistent for `engines` and `EnginesMap`.
2. The size of the `Engines` array is very little. At most three. I think the optimization is unnecessary.

Reply:
Use `set.StringSet` is better.